opencti-platform/opencti-graphql/src/modules/disseminationList/disseminationList-domain.ts

```diff
@@ -0,0 +1,87 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*/
+
+import ejs from 'ejs';
+import type { AuthContext, AuthUser } from '../../types/user';
+import { internalLoadById, listEntitiesPaginated, storeLoadById } from '../../database/middleware-loader';
+import type { DisseminationListSendInput, QueryDisseminationListsArgs } from '../../generated/graphql';
+import { type BasicStoreEntityDisseminationList, ENTITY_TYPE_DISSEMINATION_LIST } from './disseminationList-types';
+import { sendMail } from '../../database/smtp';
+import { getEntityFromCache } from '../../database/cache';
+import type { BasicStoreSettings } from '../../types/settings';
+import { SYSTEM_USER } from '../../utils/access';
+import { ENTITY_TYPE_SETTINGS } from '../../schema/internalObject';
+import { downloadFile, loadFile } from '../../database/file-storage';
+import { buildContextDataForFile, publishUserAction } from '../../listener/UserActionListener';
+import { EMAIL_TEMPLATE } from '../../utils/emailTemplates/emailTemplate';
+import { READ_DATA_INDICES, READ_INDEX_DELETED_OBJECTS } from '../../database/utils';
+import type { BasicStoreObject } from '../../types/store';
+
+export const findById = (context: AuthContext, user: AuthUser, id: string) => {
+  return storeLoadById<BasicStoreEntityDisseminationList>(context, user, id, ENTITY_TYPE_DISSEMINATION_LIST);
+};
+
+export const findAll = (context: AuthContext, user: AuthUser, args: QueryDisseminationListsArgs) => {
+  return listEntitiesPaginated<BasicStoreEntityDisseminationList>(context, user, [ENTITY_TYPE_DISSEMINATION_LIST], args);
+};
+
+interface SendMailArgs {
+  from: string;
+  to: string;
+  bcc?: string[];
+  subject: string;
+  html: string;
+  attachments?: any[];
+}
+
+export const sendToDisseminationList = async (context: AuthContext, user: AuthUser, input: DisseminationListSendInput) => {
+  const settings = await getEntityFromCache<BasicStoreSettings>(context, SYSTEM_USER, ENTITY_TYPE_SETTINGS);
+  const filePath = input.email_attached_file_id;
+  const file = await loadFile(context, user, filePath);
+  if (file && file.metaData.mimetype === 'application/pdf' && file.metaData.entity_id && settings.valid_enterprise_edition) {
+    const stream = await downloadFile(file.id);
+    const emailBodyFormatted = input.email_body.replaceAll('\n', '<br/>');
+    const generatedEmail = ejs.render(EMAIL_TEMPLATE, { settings, body: emailBodyFormatted });
+    const sendMailArgs: SendMailArgs = {
+      from: settings.platform_email,
+      to: settings.platform_email,
+      bcc: [input.email_address, user.user_email],
+      subject: input.email_object,
+      html: generatedEmail,
+      attachments: [
+        {
+          filename: file.name,
+          content: stream,
+        }
+      ],
+    };
+    await sendMail(sendMailArgs);
+    const instance = await internalLoadById(context, user, file.metaData.entity_id, { indices: [...READ_DATA_INDICES, READ_INDEX_DELETED_OBJECTS] });
```

Comment:
Why do we need deleted objects ? If a pdf is deleted, it's not possible to send it, or is it ?